tmt/steps/execute/__init__.py

```diff
@@ -34,6 +57,28 @@ def save(self):
             (result.name, result.export()) for result in self.results()])
         self.write('results.yaml', tmt.utils.dict_to_yaml(results))
 
+    def _map_old_methods(self):
+        """ Map the old execute methods in a backward-compatible way """
+        how = self.data[0]['how']
+        matched = re.search(r"^(shell|beakerlib)(\.(tmt|detach))?$", how)
+        if not matched:
+            return
+        # Show the old method deprecation warning to users
+        self.warn(f"The '{how}' execute method has been deprecated.")
```

Comment:
And will be removed by March 2021? :)

Reply:
seems quite an important piece of info

Reply:
what about adding here a check fo March 2021 and just error in the case the date has passed? This way users will be nicely informed with an error that they are using a deprecated functionality

Reply:
Well this is showing a warning message for each obsolete config already now:

```
> tmt run plan --name plans/smoke
/var/tmp/tmt/run-478

/plans/smoke
    warning: The 'shell.tmt' execute method has been deprecated.
    warning: Use 'how: tmt' in the execute step instead (L2).
    warning: Set 'framework: shell' in test metadata (L1).
    warning: Support for old methods will be dropped in March 2021.
    discover
        how: shell
        directory: /home/psss/git/tmt
        summary: 1 test selected
    provision
        how: local
        distro: Fedora 32 (Workstation Edition)
        summary: 1 guest provisioned
    prepare
        summary: 0 preparations applied
    execute
        how: tmt
        summary: 1 test executed
    report
        how: display
        summary: 1 test passed
    finish
        summary: 0 tasks completed

total: 1 test passed
```

In March I would like to completely remove the functionality and fail with an error.

Reply:
agreed, I was just proposing to add the check now .. .but it is your choice :+1:

Reply:
Also, before completely removing the feature I would rather check the current situation at that time and announce this backward incompatible change so that users are not surprised.